### PR TITLE
Dilpreet | Test | Thermal Template

### PR DESCRIPTION
**Pull request by dilpreetsinghofficial**
https://app.clickup.com/t/23m4guw

### CODE REVIEW

File: apps/web-giddh/src/app/invoice/preview/models/invoice-preview-details/invoice-preview-details.component.ts [hunk 1]
```diff
@@ -200,9 +212,17 @@ export class InvoicePreviewDetailsComponent implements OnInit, OnChanges, AfterV
                 (this.attachedDocumentType.type === 'pdf' || this.attachedDocumentType.type === 'image')));
     }
 
-    public ngOnInit(): void {
+    ngOnInit() {
```
Comment (the-prateek-mishra) [inline apps/web-giddh/src/app/invoice/preview/models/invoice-preview-details/invoice-preview-details.component.ts]: Modifier and return type is getting removed

File: apps/web-giddh/src/app/invoice/templates/edit-template/out-tempate/templates/gst-template-a/gst-template-a.component.html [hunk 2]
```diff
@@ -628,10 +628,7 @@
                             </tr>
                         </thead>
                         <tbody>
-                            <tr
-                                class="text-right"
-                                *ngIf="fieldsAndVisibility.table.previousDue?.display && voucherType === 'sales'"
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/invoice/templates/edit-template/out-tempate/templates/gst-template-a/gst-template-a.component.html]: fix this

File: apps/web-giddh/src/app/invoice/templates/edit-template/out-tempate/templates/gst-template-a/gst-template-a.component.scss [hunk 3]
```diff
@@ -1,3 +1,5 @@
+@import "../../../../../../../assets/css/color.scss";
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/invoice/templates/edit-template/out-tempate/templates/gst-template-a/gst-template-a.component.scss]: remove this

File: apps/web-giddh/src/app/invoice/templates/edit-template/out-tempate/templates/gst-template-a/gst-template-a.component.scss [hunk 4]
```diff
@@ -535,7 +537,7 @@ table.tax-table.item {
 }
 .warehouse-table {
     width: 100%;
-    border-bottom: 1px solid var(--color-table-primary-content-bar-border);
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/invoice/templates/edit-template/out-tempate/templates/gst-template-a/gst-template-a.component.scss]: revert this

File: apps/web-giddh/src/app/invoice/templates/edit-template/out-tempate/templates/thermal-template/thermal-template.component.ts [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+    /** This will use for company settings */
+    public companySetting$: Observable<any> = observableOf(null);
+    /** This will use for column visibililty */
+    public columnsVisibled: number;
+    /** This will use for on destroy component */
+    private destroyed$: ReplaySubject<boolean> = new ReplaySubject(1);
+
+    constructor(private store: Store<AppState>,
+        private settingsProfileActions: SettingsProfileActions) {
+        this.companySetting$ = this.store.pipe(select(s => s.settings.profile), takeUntil(this.destroyed$));
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/invoice/templates/edit-template/out-tempate/templates/thermal-template/thermal-template.component.ts]: don't use single character variable

File: apps/web-giddh/src/app/invoice/templates/edit-template/out-tempate/templates/thermal-template/thermal-template.component.ts [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+    }
+
+    /**
+     * Lifecycle hook use for initialization
+     *
+     * @memberof ThermalTemplateComponent
+     */
+    public ngOnInit() : void {
+        this.companySetting$.subscribe(a => {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/invoice/templates/edit-template/out-tempate/templates/thermal-template/thermal-template.component.ts]: don't use single character variable

File: apps/web-giddh/src/app/services/thermal.service.ts [hunk 7]
```diff
@@ -0,0 +1,721 @@
...
+                "\x30" +
+                // <!-- END QR DATA -->
+                this.printerFormat.lineBreak +
+                this.justifyText(thankYouMsgField, footerCompanyName) +
+                this.printerFormat.lineBreak + this.printerFormat.lineBreak +
+                this.justifyText(firmNameField, "");
+            qz.websocket
+                .connect()
+                .then(function () {
+                    return qz.printers.find("Rugtek"); // Pass the printer name into the next Promise
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/services/thermal.service.ts:651]: this printer name should be dynamic once certificate is generated

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

**Review by ravinderwalkover — APPROVED** [review]

**Review by the-prateek-mishra — APPROVED** [review]

File: package.json [hunk 8]
```diff
@@ -124,7 +124,9 @@
     "ngx-toastr": "13.0.0",
     "ngx-uploader": "11.0.0",
     "node": "^14.17.2",
+    "path-browserify": "^1.0.1",
```
Comment (the-prateek-mishra) [inline package.json:127]: Please check this

File: tsconfig.json [hunk 9]
```diff
@@ -35,6 +36,9 @@
             ],
             "@giddh-workspaces/web/*": [
                 "xplat/web/*"
+            ],
+            "path": [
```
Comment (the-prateek-mishra) [inline tsconfig.json]: Please check this

**Review by the-prateek-mishra — CHANGES_REQUESTED** [review]

**Review by the-prateek-mishra — APPROVED** [review]

**Pull request merged**